ci/test_cudf_polars.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -eou pipefail
+
+# We will only fail these tests if the PR touches code in pylibcudf
+# or cudf_polars itself.
```

Comment:
Is the goal of this just to ensure that `polars` pushing a new release candidate doesn't break these tests, and just for this transitory period where `polars` 1.0 is not released yet?

If so, I think it'd be a lot simpler and more reliable to just `==` pin to a specific release of `polars` (at test time). So like having `cudf_polars` depend on `polars>=1.0.0a0` and then having its `[test]` extra depend on `polars==1.0.0rc2`.

At test time, that `==` will ensure you get exactly the version you want, and then you could be confident that any failing tests are not the result of a new `polars` release.

And then once there's an official release of `polars`, hopefully you'd be able to remove that pin in the `[test]` extra entirely and switch to only uses *releases* of `polars` (like `polars>=1.0.0`).

Reply:
> Is the goal of this just to ensure that polars pushing a new release candidate doesn't break these tests, and just for this transitory period where polars 1.0 is not released yet?

Although we expect going forward (post 1.0) that there will be infrequent changes in the API we use, it is not ruled out (the interfaces we're using are not part of the public API, so not subject to the semantic versioning constraints polars has). So the goal here is that changes that don't touch cudf-polars do not block CI for everything else (due to upstream changes).

So it's both "for now", but also going forward.